R/derive_locf_records.R

```diff
@@ -0,0 +1,184 @@
+#'
+#'   For each group defined by `by_vars` an observation is added to the output
+#'   dataset. Only variables specified in `by_vars` will be populated
+#'   in the newly created records.
+#'
+#' @param order List of variables for sorting a dataset
+#'
+#'   The dataset is sorted by `order` before carrying the last
+#'   observation (eg. `AVAL`) within each `by_vars` forward.
+#'
+#' @author G Gayatri
+#'
+#' @details For each group (with respect to the variables specified for the
+#'   `by_vars` parameter) an observation is added to the output dataset from
+#'   `dataset_expected_obs`, if the input dataset does not contain observations
+#'   for missed visits/time points or if AVAL is missing.
+#'
+#'   For the new observations, `AVAL` is set to the non-missing `AVAL` of the
+#'   previous observation in the input dataset (when sorted by `order`) and
+#'   `DTYPE` is set to `LOCF`.
+#'
+#' @return The input dataset with the new `LOCF` observations added for each
+#' `by_vars`. Note, a variable will only be populated in the new parameter rows
+#' if it is specified in `by_vars`.
+#'
+#' @keywords der_bds_findings
+#' @family der_bds_findings
+#'
+#' @export
+#'
+#' @examples
+#'
+#' advs <- tibble::tribble(
+#'   ~STUDYID,  ~USUBJID,      ~PARAMCD, ~PARAM,                          ~AVAL, ~AVISITN, ~AVISIT,
+#'   "CDISC01", "01-701-1015", "DIABP", "Diastolic Blood Pressure (mmHg)", 51,    0,       "BASELINE",
+#'   "CDISC01", "01-701-1015", "DIABP", "Diastolic Blood Pressure (mmHg)", 50,    2,       "WEEK 2",
+#'   "CDISC01", "01-701-1015", "DIABP", "Diastolic Blood Pressure (mmHg)", 51,    4,       "WEEK 4",
+#'   "CDISC01", "01-701-1015", "DIABP", "Diastolic Blood Pressure (mmHg)", 50,    6,       "WEEK 6",
+#'   "CDISC01", "01-701-1015", "SYSBP", "Systolic Blood Pressure (mmHg)", 121,    0,       "BASELINE",
+#'   "CDISC01", "01-701-1015", "SYSBP", "Systolic Blood Pressure (mmHg)", 121,    2,       "WEEK 2",
+#'   "CDISC01", "01-701-1015", "SYSBP", "Systolic Blood Pressure (mmHg)", 121,    4,       "WEEK 4",
+#'   "CDISC01", "01-701-1015", "SYSBP", "Systolic Blood Pressure (mmHg)", 121,    6,       "WEEK 6",
+#'   "CDISC01", "01-701-1028", "DIABP", "Diastolic Blood Pressure (mmHg)", 79,    0,       "BASELINE",
+#'   "CDISC01", "01-701-1028", "DIABP", "Diastolic Blood Pressure (mmHg)", 80,    2,       "WEEK 2",
+#'   "CDISC01", "01-701-1028", "DIABP", "Diastolic Blood Pressure (mmHg)", NA,    4,       "WEEK 4",
+#'   "CDISC01", "01-701-1028", "DIABP", "Diastolic Blood Pressure (mmHg)", NA,    6,       "WEEK 6",
+#'   "CDISC01", "01-701-1028", "SYSBP", "Systolic Blood Pressure (mmHg)", 130,    0,       "BASELINE",
+#'   "CDISC01", "01-701-1028", "SYSBP", "Systolic Blood Pressure (mmHg)", 132,    2,       "WEEK 2"
+#' )
+#'
+#'
+#' # A dataset with all the combinations of PARAMCD, PARAM, AVISIT, AVISITN, ... which are expected.
+#' advs_expected_obsv <- tibble::tribble(
+#'  ~PARAMCD, ~PARAM, ~AVISITN, ~AVISIT,
```

Comment:
If the expected visits do not depend on the parameter, you do not need to include the parameter in the dataset. I would either remove the parameter or make the visits parameter dependent.

Reply:
Made the visits parameter dependent.